src/libraries/Common/src/Interop/Linux/os-release/Interop.OSReleaseFile.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+
+internal static partial class Interop
+{
+    // Parse information from '/etc/os-release'.
+    internal static class OSReleaseFile
+    {
+        private const string EtcOsReleasePath = "/etc/os-release";
+
+        /// <summary>
+        /// Returns a user-friendly distribution name.
+        /// </summary>
+        internal static string? GetPrettyName(string filename = EtcOsReleasePath)
+        {
+            if (File.Exists(filename))
+            {
+                string[] lines;
+                try
+                {
+                    lines = File.ReadAllLines(filename);
+                }
+                catch
+                {
+                    return null;
+                }
+
+                // Parse the NAME, PRETTY_NAME, and VERSION fields.
+                // These fields are suitable for presentation to the user.
+                ReadOnlySpan<char> prettyName = default, name = default, version = default;
+                foreach (var line in lines)
+                {
+                    ReadOnlySpan<char> lineSpan = line.AsSpan();
+
+                    _ = TryGetFieldValue(lineSpan, "PRETTY_NAME=", ref prettyName) ||
+                        TryGetFieldValue(lineSpan, "NAME=", ref name) ||
+                        TryGetFieldValue(lineSpan, "VERSION=", ref version);
+                }
```

Comment:
How many lines are typically in this file?  Wondering if we should be breaking out of it once prettyName is non-empty?

Reply:
Up to 25 lines.

Do you want me to add the break?

Reply:
If it's likely the pretty name will come early, yeah, let's add it.  Otherwise, it's fine as is.